skyportal/handlers/api/annotation.py

```diff
@@ -8,13 +8,32 @@
 
 class AnnotationHandler(BaseHandler):
     @auth_or_token
-    def get(self, annotation_id):
+    def get(self, associated_resource_type, resource_id, annotation_id):
         """
         ---
         description: Retrieve an annotation
         tags:
           - annotations
         parameters:
+          - in: path
+            name: associated_resource_type
+            required: true
+            schema:
+              type: string
+            description: |
+               What underlying data the annotation is on:
+               an "object" (default), or a "spectrum".
+          - in: path
+            name: resource_id
+            required: true
+            schema:
+              type: string
+            description: |
+               The ID of the underlying data.
+               This would be a string for an object ID
```

Comment:
I'm confused that objects are still mentioned here, when we are only handling sources and spectra.  If you think this is the best language we'll leave it as is; but it feels confusing to have `/api/sources/...` and then talk about an object, which is more of an internal SkyPortal concept.